### PR TITLE
feat: allow third party mail server

### PR DESCRIPTION
**Pull request by saranshisatgit**
Not sure this is the right approach since gmail services  is  default so this creates if the SendGrid is provided as the smtp settings !

### CODE REVIEW

File: packages/backend/server/src/modules/auth/mailer/mailer.ts [hunk 1]
```diff
@@ -26,6 +26,17 @@ export const MAILER: FactoryProvider<
         },
       });
     }
+  // Create a transport for sendgrid 
+    if(config.auth.email.server == 'smtp.sendgrid.net'){
+      return createTransport({
+        host: config.auth.email.server,
+        port: 587,
```
Comment (darkskygit) [inline packages/backend/server/src/modules/auth/mailer/mailer.ts]: can use `config.auth.email.port`

File: packages/backend/server/src/modules/auth/mailer/mailer.ts [hunk 2]
```diff
@@ -26,6 +26,17 @@ export const MAILER: FactoryProvider<
         },
       });
     }
+  // Create a transport for sendgrid 
+    if(config.auth.email.server == 'smtp.sendgrid.net'){
```
Comment (darkskygit) [inline packages/backend/server/src/modules/auth/mailer/mailer.ts]: can check if server not equal `smtp.gmail.com`

**Review by darkskygit — CHANGES_REQUESTED** [review]

**Comment by saranshisatgit** [comment]
@darkskygit This one could be merged thanks !

**Comment by saranshisatgit** [comment]
hey @darkskygit any updates?

**Comment by darkskygit** [comment]
this problem was fix by #5835 , thanks for your PR

Now you can use the following envs to define the email sender:
``` env
MAILER_HOST=
MAILER_PORT=
MAILER_USER=
MAILER_PASSWORD=
MAILER_SENDER=
```

**Pull request closed without merging**